R/format_PFN.R

```diff
@@ -336,216 +517,407 @@ create_brood_EDM <- function(Primary_data, ReRingTable){
+        TRUE ~ NA_character_
+      ),
+
+      # 3.2) Species
+      #      (Set to 'CCCCCC' if conflicting)
+      Species = dplyr::case_when(
+        is.na(.data$Species_Nest) ~ .data$Species_IPMR,
+        is.na(.data$Species_IPMR) ~ .data$Species_Nest,
+        .data$Species_Nest == .data$Species_IPMR  ~ .data$Species_Nest,
+        .data$Species_Nest != .data$Species_IPMR ~ 'CCCCCC'
+      ),
+      # MB: These should be flagged as conflicted and checked manually.
+
+      # 3.3) LocationID
+      #     (Use Nest if conflicting and there is no unconfirmed Nest match)
+      #     (Use IPMR if conflicting and there is an unconfirmed Nest match)
+      LocationID = dplyr::case_when(
+        is.na(.data$LocationID_Nest) ~ .data$LocationID_IPMR,
+        is.na(.data$LocationID_IPMR) ~ .data$LocationID_Nest,
+        .data$LocationID_Nest == .data$LocationID_IPMR ~ .data$LocationID_Nest,
+        .data$LocationID_Nest != .data$LocationID_IPMR & !.data$UnconfirmedNestMatch ~ .data$LocationID_Nest,
+        .data$LocationID_Nest != .data$LocationID_IPMR & .data$UnconfirmedNestMatch ~ .data$LocationID_IPMR
+      ),
+      # MB: Generally Nest data priority here (spent a lot of time cleaning this in Nest files).
+      #     Currently exception for conflicts in unconfirmed matches: IPMR priority
+      #     (We may revisit the latter decision at a later point)
+
+
+      # 3.4) CapturePopID & ReleasePopID
+      CapturePopID = dplyr::case_when(
+        is.na(.data$CapturePopID_Nest) ~ .data$CapturePopID_IPMR,
+        is.na(.data$CapturePopID_IPMR) ~ .data$CapturePopID_Nest,
+        .data$CapturePopID_Nest == .data$CapturePopID_IPMR ~ .data$CapturePopID_Nest,
+        .data$CapturePopID_Nest != .data$CapturePopID_IPMR ~ .data$CapturePopID_Nest
+      ),
+      ReleasePopID = dplyr::case_when(
+        is.na(.data$ReleasePopID_Nest) ~ .data$ReleasePopID_IPMR,
+        is.na(.data$ReleasePopID_IPMR) ~ .data$ReleasePopID_Nest,
+        .data$ReleasePopID_Nest == .data$ReleasePopID_IPMR ~ .data$ReleasePopID_Nest,
+        .data$ReleasePopID_Nest != .data$ReleasePopID_IPMR ~ .data$ReleasePopID_Nest
+      ),
+      # NOTE: There are no occurrences of conflicts here in the data at the moment.
+
+      # 3.5) CapturePlot & ReleasePlot
+      CapturePlot = dplyr::case_when(
+        is.na(.data$CapturePlot_Nest) ~ .data$CapturePlot_IPMR,
+        is.na(.data$CapturePlot_IPMR) ~ .data$CapturePlot_Nest,
+        .data$CapturePlot_Nest == .data$CapturePlot_IPMR ~ .data$CapturePlot_Nest,
+        .data$CapturePlot_Nest != .data$CapturePlot_IPMR ~ .data$CapturePlot_Nest
+      ),
+      ReleasePlot = dplyr::case_when(
+        is.na(.data$ReleasePlot_Nest) ~ .data$ReleasePlot_IPMR,
+        is.na(.data$ReleasePlot_IPMR) ~ .data$ReleasePlot_Nest,
+        .data$ReleasePlot_Nest == .data$ReleasePlot_IPMR ~ .data$ReleasePlot_Nest,
+        .data$ReleasePlot_Nest != .data$ReleasePlot_IPMR ~ .data$ReleasePlot_Nest
+      ),
```

Comment:
This rule could be written out as a single function so it doesn't need to be retyped each time. From a quick check it seems you're using the same decision rule every time (i.e. use Nest data where possible)

Reply:
Most of the time, but not always. Priority is given to IPMR for example for `CaptureAlive` and `ReleaseAlive`, and for some columns, information is consolidated differently. 

(Not quite sure how I would write it as a single function. Without actually writing a function...)

Reply:
So just something like this would work, and you could add some variation to work for other columns

```
compare_IPMR_Nest <- function(Nestcol, IPMRcol){
         
             dplyr::case_when(is.na({{Nestcol}}) ~ {{IPMRcol}},
                                                     is.na({{IPMRcol}}) ~ {{Nestcol}},
                                                     {{Nestcol}} == {{IPMRcol}} ~ {{Nestcol}},
                                                     {{Nestcol}} != {{IPMRcol}} ~ {{Nestcol}})
}

#Now this would work. Saves 5 lines of code
Capture_data %>%
mutate(CapturePopID = compare_IPMR_Nest(.data$CapturePopID_Nest, .data$CapturePopID_IPMR)
```